README.md
# Supportik
Описание:
Утилита, написанная на языке программирования Python, которая должна облегчить работу всем, кто работает с доменами, сайтами и не только. Недохакерам тоже буду рад.
Description:
A utility written in the Python programming language, which should make the work of everyone who works with domains, websites and more easier.

Утилита иногда будет обновляться, постепенно увеличивая количество возможностей. Для того, чтобы узнать список команд и получить пояснение к ним, введите в утилите команду "help". Приложен журнал изменений в версиях в файле changelog.


Changelog.txt
Версия 1.0: [12.7.2022]
Добавлен вступительный текст
Добавлена команда getip
Добавлена команда getdhcp
Добавлена команда help
Добавлена команда getpcname
Добавлена команда getlocalip
Добавлена команда getdomainip

Версия 1.1 [15.7.2022]
Теперь домен для getdomainip нужно вводить через пробел
Добавлена команда sitestatus
Обновлена команда help
Оптимизация

Версия 1.2 [16.7.2022]
Добавлена команда speedtest
Добавлена команда getmac
Добавлена команда getmacnum
Изменён текст при ошибке
Добавлена команда scanportsonip
Добавлена команда scanportsondomain
Порты проверяются в диапазоне 1-1000

Версия 1.3 [17.7.2022]
Добавлена возможность писать команды на русском
Заменён начальный текст
В help добавлены команды на русском
Добавлена команда genpass / сделатьпароль
Исправлены баги с появившимися русскими командами
Проверка портов ускорена в 8 раз (~8,145)
Добавлена команда rangeportsdomain / диапазонпортовдомен
Добавлена команда rangeportsip / диапазонпортовайпи
Теперь старые команды проверки портов работают по списку
У проверок диапазона теперь можно указать диапазон
Оптимизирован код
Добавлен список портов для проверки (7, 20, 21, 22, 23, 25, 69, 80, 443, 110, 143, 389, 3306, 5432)

Версия 1.4 [18.7.2022]
Проверке по домену добавлен список портов проверки по IP
Сканирование диапазона портов слегка замедленно чтобы не пропускать порты
Добавлены порты для проверки (8083, 993, 995, 53, 8080, 3128, 2222, 2083, 2087, 587, 81, 115)
Обновлён порядок портов для проверки
Проверка открытых популярных портов ускорена на ~10,59%
Добавлена команда getcoords / моикоординаты
Добавлена команда coordsbyip / координатыайпи
Добавлена команда citybyip / городпоайпи
Русская команда "получитьайпи" заменена на "мойайпи"

Версия 1.5 [19.7.2022]
Сканер портов по диапазону больше не пропускает открытые порты
Добавлена команда genpassv2 / сложныйпароль
Старые команды пароля добавлены в help
Добавлена команда shortlink / сжатьссылку

Версия 1.6 [21.7.2022]
Исправлен баг с получением MAC-адреса
Исправлены некоторые ошибки в тексте
Сжатие ссылок удалено

supportik.py
import socket
print("""
Supportik v1.6 [21.07.2022]
Автор: Данил Пистолетов
Сайт: danil-pistoletov.org
GitHub: github.com/danilpistoletov/supportik
""")

def getip():
    import urllib.request
    external_ip = urllib.request.urlopen('https://ident.me').read().decode('utf8')
    print(external_ip)

def getdhcp():
    print(socket.gethostbyname(socket.gethostname()))

def gethelp():
    print("""
    Соблюдайте регистр команд и всё будет хорошо
    help / помощь - список команд и пояснение к ним
    getip / мойайпи - получение внешнего адреса компьютера
    getdhcp / получитьdhcp - получение вашего DHCP
    getpcname / имякомпа - получение имени вашего компьютера
    getdomainip / айписайта [ДОМЕН] - получение IP сайта (без http/https)
    sitestatus / статуссайта [ДОМЕН] - проверка статуса сайта (без http/https)
    speedtest / узнатьскорость - измерение скорости скачивания и загрузки
    getmac / получитьмак - получить MAC-адрес
    getmacnum / получитьмакцифры- получить MAC-адрес в виде числа
    scanportsonip / проверитьпортыайпи - проверить популярные открытые порты по IP
    scanportsondomain / проверитьпортыдомен - проверить популярные открытые порты по домену (без http/https)
    rangeportsdomain / диапазонпортовдомен - проверить открытые порты по домену (без http/https)
    rangeportsip / диапазонпортовайпи - проверить открытые порты по айпи
    getcoords / моикоординаты - узнать координаты по своему IP-адресу
    coordsbyip / координатыайпи - узнать координаты по указанному IP-адресу
    citybyip / городпоайпи - узнать город по IP-адресу
    genpass / сделатьпароль - генерирует пароль (английский алфавит + цифры)
    genpassv2 / сложныйпароль - генерирует пароль (английский и русский алфавиты + цифры + символ)
    """)

def getpcname():
    print(socket.getfqdn())

def getlocalip():
    host = socket.getaddrinfo(socket.gethostname(), None)
    ipv4_addresses = [i[4][0] for i in host if i[0] == socket.AF_INET]
    print(ipv4_addresses)

def getdomainip(x):
    print(socket.gethostbyname(x))

def sitestatus(x):
    import requests
    import urllib.request
    from urllib.error import URLError
    try:
        urllib.request.urlopen("https://" + x)
        print("Сайт доступен")
        status = requests.get('https://' + x)
        print("Код сайта: ", status.status_code)
    except URLError:
        print("Сайт недоступен")

def sspeedtest():
    import speedtest
    st = speedtest.Speedtest()
    download = round(st.download() / 1048576)
    upload = round(st.upload() / 1048576)
    print("Скачивание: ", download, "МБ/С | " "Загрузка: ", upload, "МБ/С")

def getmac():
    from uuid import getnode
    import re
    print(':'.join(re.findall('..', '%012x' % getnode())))

def getmacnum():
    from uuid import getnode
    print(getnode())

def scanports(ip):
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.settimeout(0.08)
    try:
        connect = sock.connect((ip, i))
        print("Порт ", i, " открыт")
        sock.close()
    except:
        pass

def genpass():
    import random
    variable = "1234567890qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM"
    length = int(input("Введите длину пароля\n"))
    password = ""
    for i in range(length):
        password += random.choice(variable)
    print(password)

def genpassv2():
    import random
    variable = "йцукенгшщзхъфывапролджэячсмитьбюЙЦУКЕНГШЩЗХЪФЫВАПРОЛДЖЭЯЧСМИТЬБЮ1234567890qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM!@#$%^&*/~"
    length = int(input("Введите длину пароля\n"))
    password = ""
    for i in range(length):
        password += random.choice(variable)
    print(password)

def scanportsv2(ip):
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.settimeout(0.08)
    try:
        connect = sock.connect((ip, i))
        print("Порт ", i, " открыт")
        sock.close()
    except:
        pass

while "anime" == "anime":
    command = input()
    if command == "getip" or command == "мойайпи":
        getip()
    elif command == "getdhcp" or command == "получитьdhcp":
        getdhcp()
    elif command == "help" or command == "помощь":
        gethelp()
    elif command == "getpcname" or command == "имякомпа":
        print(socket.getfqdn())
    elif command == "getlocalip":
        getlocalip()
    elif "getdomainip" in command:
        domain = command[12:]
        getdomainip(domain)
    elif "айписайта" in command:
        domain1 = command[10:]
        getdomainip(domain1)
    elif "sitestatus" in command:
        siteforcheck = command[11:]
        sitestatus(siteforcheck)
    elif "статуссайта" in command:
        siteforcheck1 = command[12:]
        sitestatus(siteforcheck1)
    elif command == "speedtest" or command == "узнатьскорость":
        sspeedtest()
    elif command == "getmac" or command == "получитьмак":
        getmac()
    elif command == "getmacnum" or command == "получитьмакцифры":
        getmacnum()
    elif "rangeportsip" in command:
        na4alo = int(input("Введите начальный порт\n"))
        predel = int(input("Введите последний порт\n"))
        for i in range(na4alo, predel):
            scanports(command[13:])
    elif "диапазонпортовайпи" in command:
        na4alo1 = int(input("Введите начальный порт\n"))
        predel1 = int(input("Введите последний порт\n"))
        for i in range(na4alo1, predel1):
            scanports(command[19:])
    elif "rangeportsdomain" in command:
        na4alo2 = int(input("Введите начальный порт\n"))
        predel2 = int(input("Введите последний порт\n"))
        for i in range(na4alo2, predel2):
            scanports(socket.gethostbyname(command[17:]))
    elif "диапазонпортовдомен" in command:
        na4alo3 = int(input("Введите начальный порт\n"))
        predel3 = int(input("Введите последний порт\n"))
        for i in range(na4alo3, predel3):
            scanports(socket.gethostbyname(command[20:]))
    elif "проверитьпортыдомен" in command:
        ports = [7, 20, 21, 22, 23, 25, 53, 69, 80, 81, 110, 115, 143, 389, 443, 587, 993, 995, 2083, 2087, 2222, 3128, 3306, 5432, 8080, 8083]
        for i in ports:
            scanportsv2(socket.gethostbyname(command[20:]))
    elif "scanportsondomain" in command:
        ports = [7, 20, 21, 22, 23, 25, 53, 69, 80, 81, 110, 115, 143, 389, 443, 587, 993, 995, 2083, 2087, 2222, 3128, 3306, 5432, 8080, 8083]
        for i in ports:
            scanportsv2(socket.gethostbyname(command[19:]))
    elif "проверитьпортыайпи" in command:
        ports = [7, 20, 21, 22, 23, 25, 53, 69, 80, 81, 110, 115, 143, 389, 443, 587, 993, 995, 2083, 2087, 2222, 3128, 3306, 5432, 8080, 8083]
        for i in ports:
            scanportsv2(command[19:])
    elif "scanportsonip" in command:
        ports = [7, 20, 21, 22, 23, 25, 53, 69, 80, 81, 110, 115, 143, 389, 443, 587, 993, 995, 2083, 2087, 2222, 3128, 3306, 5432, 8080, 8083]
        for i in ports:
            scanportsv2(command[14:])
    elif command == "genpass" or command == "сделатьпароль":
        genpass()
    elif command == "getcoords" or command == "моикоординаты":
        import geocoder
        g = geocoder.ipinfo("me")
        print(g.latlng)
    elif "coordsbyip" in command:
        import geocoder
        g1 = geocoder.ipinfo(command[11:])
        print(g1.latlng)
    elif "координатыайпи" in command:
        import geocoder
        g2 = geocoder.ipinfo(command[15:])
        print(g2.latlng)
    elif "citybyip" in command:
        import geocoder
        g3 = geocoder.ipinfo(command[9:])
        print(g3.city)
    elif "городпоайпи" in command:
        import geocoder
        g4 = geocoder.ipinfo(command[12:])
        print(g4.city)
    elif command == "genpassv2" or command == "сложныйпароль":
        genpassv2()
    else:
        print("Неверная команда. Вы можете ввести \"помощь\" для просмотра всех команд")